docs/edgeql/group.rst

```diff
@@ -0,0 +1,195 @@
+    ... group M{title}
+    ... using decade := .release_year // 10
+    ... by decade;
+    {
+      {
+        key: {decade: 201},
+        grouping: {'decade'},
+        elements: {
+          default::Movie {title: 'Captain America: Civil War'},
+          default::Movie {title: 'Doctor Strange'},
+          default::Movie {title: 'Spider-Man: Homecoming'},
+          default::Movie {title: 'Thor: Ragnarok'},
+          default::Movie {title: 'Ant-Man and the Wasp'},
+          default::Movie {title: 'Spider-Man: No Way Home'},
+        },
+      },
+      {
+        key: {decade: 202},
+        grouping: {'decade'},
+        elements: {default::Movie {title: 'Black Widow'}},
+      },
+    }
+
+It's also possible to group by more than one parameter, so we can group by
+whether the movie ``title`` contains a colon *and* the decade it was released:
+
+.. code-block:: edgeql-repl
+
+    db> with M := (select Movie filter .release_year > 2015)
+    ... group M{title}
+    ... using
+    ...   decade := .release_year // 10,
+    ...   has_colon := .title like '%:%'
+    ... by decade, has_colon;
+    {
+      {
+        key: {decade: 201, has_colon: false},
+        grouping: {'decade', 'has_colon'},
+        elements: {
+          default::Movie {title: 'Ant-Man and the Wasp'},
+          default::Movie {title: 'Doctor Strange'},
+        },
+      },
+      {
+        key: {decade: 201, has_colon: true},
+        grouping: {'decade', 'has_colon'},
+        elements: {
+          default::Movie {title: 'Captain America: Civil War'},
+          default::Movie {title: 'Spider-Man: No Way Home'},
+          default::Movie {title: 'Thor: Ragnarok'},
+          default::Movie {title: 'Spider-Man: Homecoming'},
+        },
+      },
+      {
+        key: {decade: 202, has_colon: false},
+        grouping: {'decade', 'has_colon'},
+        elements: {default::Movie {title: 'Black Widow'}},
+      },
+    }
+
+Once we break a set into partitions, we can also use :ref:`aggregate
+<ref_eql_set_aggregate>` functions to provide some analytics about the data.
+For example, for the above partitioning (by decade and presence of ``:`` in
+the ``title``) we can calculate how many movies are in each subset as well as
+the average number of words in the movie titles:
+
+.. code-block:: edgeql-repl
+
+    db> with M := (select Movie filter .release_year > 2015)
+    ... select (
+    ...   group M{title}
+    ...   using
+    ...     decade := .release_year // 10 - 200,
+    ...     has_colon := .title like '%:%'
+    ...   by decade, has_colon
+    ... ) {
+    ...   key := .key{decade, has_colon},
+    ...   count := count(.elements),
+    ...   avg_words := math::mean(
+    ...     len(str_split(.elements.title, ' ')))
+    ... };
+    {
+      {key: {decade: 1, has_colon: false}, count: 2, avg_words: 3},
+      {key: {decade: 1, has_colon: true}, count: 4, avg_words: 3},
+      {key: {decade: 2, has_colon: false}, count: 1, avg_words: 2},
+    }
+
```

Comment:
I like that this page doesn't get into grouping sets but we should probably include a ``.. note::`` tag mentioning that more sophisticated things are possible with a link to the reference docs.